### PR TITLE
Handle combined audio derivatives in orphan_s3_derivatives checker.

### PR DESCRIPTION
**Pull request by eddierubeiz**
Ref #1140, which is part of #864.

- No longer flags combined audio derivatives as orphans.
- Correctly detects and deletes orphaned combined audio derivatives.
- Includes a test for the orphan derivative checker.

### CODE REVIEW

File: app/services/orphan_s3_derivatives.rb [hunk 1]
```diff
@@ -121,9 +145,38 @@ def parse_s3_path(s3_path)
   end
 
 
-  def orphaned?(asset_id, derivative_key, shrine_id)
-    return true unless asset_id.present? && derivative_key.present?
+  # Combined audio derivatives are stored at an s3_path of:
+  # combined_audio_derivatives/9d3114be-fb66-43a8-9c25-2600aa4ea7b0/combined_be68ab9e9d1ef65e8f4bdf38586db718.webm
+  # The first chunk is parsed by parse_s3_path as the 'asset_id'.
+  def combined_audio_derivative?(asset_id)
```
Comment (eddierubeiz) [inline app/services/orphan_s3_derivatives.rb:151]: this doesn't have to use `asset_id` as the argument name. Will fix.

File: app/services/orphan_s3_derivatives.rb [hunk 2]
```diff
@@ -121,9 +145,38 @@ def parse_s3_path(s3_path)
   end
 
 
-  def orphaned?(asset_id, derivative_key, shrine_id)
-    return true unless asset_id.present? && derivative_key.present?
+  # Combined audio derivatives are stored at an s3_path of:
```
Comment (eddierubeiz) [inline app/services/orphan_s3_derivatives.rb:148]: This comment is confusing. What I meant is that s3 paths for c.a.d's start in an invariable string `combined_audio_derivatives`.

**Comment by jrochkind** [comment]
I ran this on staging. Even going so far to run the `delete` version. It seemed to work right. 15K orphaned derivatives found, we don't know why there are quite so many on staging, but figuring that out is not this ticket. I spot checked some to make sure they really were orpahned, all checked out. No combined derivs showed up as orphaned as far as I could tell scanning thorugh the list.

**Comment by eddierubeiz** [comment]
No, you definitely removed some combined derivs.
Compare [production](https://s3.console.aws.amazon.com/s3/buckets/scihist-digicoll-production-derivatives?prefix=combined_audio_derivatives%2F0e84c870-0307-4714-99a0-1bde3134cf6f%2F&region=us-east-1&showversions=false#) and [staging](https://s3.console.aws.amazon.com/s3/buckets/scihist-digicoll-staging-derivatives?prefix=combined_audio_derivatives%2F0e84c870-0307-4714-99a0-1bde3134cf6f%2F&region=us-east-1&showversions=false#).

**Comment by eddierubeiz** [comment]
Going to merge this after the tests pass.

**Comment by jrochkind** [comment]
> No, you definitely removed some orphaned combined derivs.
Compare production and staging.

Huh, I scanned the logs of what was being removed, and didn't notice any. But I guess you must be right? Anyway, fair enough -- it's hard to keep track of exactly what's going on with 200K files.

**Pull request merged**